### DISCUSSION TITLE
Do developers consider adding the function of averaging field within specified time intervals in the output section ？

### DISCUSSION THREAD

**Discussion by ercoop** [BLAST-WarpX/warpx#4563] [Data Analysis]
For better reducing noise to analyze static fields, and  meanwhile reducing the amount of output data.

**Comment by dpgrote** [comment]
The field probe diagnostics can be integrated in time. They are described in this section of the documentation https://warpx.readthedocs.io/en/latest/usage/parameters.html#reduced-diagnostics. The integration is done every time step even when the data is written out less frequently.

**Comment by ax3l** [comment]
That is a great question, yes, besides the field probe temporal (and, as a quick helper, spatial) averaging over long averaging windows are super helpful.

Although this can already be added via Python (add a callback in each step, average down multiple steps in a custom multi-fab), we could need this for both diagnostics as well as some physical processes (e.g., Keldysh ionization).

Time-averaging is not trivial, because one needs to hold copies of the data, which can quickly outrun the available memory if the output interval is shorter than the average interval. Spatial averaging can be used as a good work-around quite often. Both features would be good additions.

For now, you can already use "coarsening" in our diagnostics, if a lower-resolution spatial average is ok for your purpose:
https://warpx.readthedocs.io/en/latest/usage/parameters.html#full-diagnostics
> <diag_name>.coarsening_ratio (list of int) optional (default 1 1 1)
>
>    Reduce size of the field output by this ratio in each dimension. (This is done by averaging the field over 1 or 2 points along each direction, depending on the staggering).  [...]

**Reply by ercoop** [reply]
Thank you for your reply！
